src/with_ladj.jl

```diff
@@ -76,17 +76,30 @@ export with_logabsdet_jacobian
 end
 
 
-@inline _get_y(y_with_ladj::NTuple{2,Any,}) = y_with_ladj[1]
-@inline _get_ladj(y_with_ladj::NTuple{2,Any}) = y_with_ladj[2]
-
-_with_ladj_on_mapped(map_or_bc::Function, y_with_ladj::Tuple{Any,Real}) = y_with_ladj
+function _with_ladj_on_mapped(map_or_bc::F, y_with_ladj::Tuple{Any,Real})  where {F<:Union{typeof(map),typeof(broadcast)}}
+    return y_with_ladj
+end
 
-function _with_ladj_on_mapped(map_or_bc::Function, y_with_ladj)
-    y = map_or_bc(_get_y, y_with_ladj)
-    ladj = sum(map_or_bc(_get_ladj, y_with_ladj))
+function _with_ladj_on_mapped(map_or_bc::F, y_with_ladj) where {F<:Union{typeof(map),typeof(broadcast)}}
+    y = map_or_bc(first, y_with_ladj)
+    ladj = sum(Broadcast.instantiate(Broadcast.broadcasted(last, y_with_ladj)))
```

Comment:
I'm sure I suggested it but actually here the broadcasting machinery is not needed, it should be fine to just use
```suggestion
    ladj = sum(last, y_with_ladj)
```
This also performs pairwise summation:
```julia
julia> x = vcat((1f0, 1f0), fill((1f-8, 1f-8), 10^8));

julia> sum(last, x)
1.9999996f0
```
The `Broadcast.instantiate` "trick" is only relevant when summing over iterators which can occur e.g. when one `zip`s two collections:
```julia
julia> y = rand(10^8 + 1);

julia> sum(last ∘ first, zip(x, y))
1.0f0

julia> sum(Broadcast.instantiate(Broadcast.broadcasted((x, y) -> last(x), x, y)))
1.9999996f0
```

Reply:
I'll benchmark it.

Reply:
No difference in benchmark and code it simpler, let's take your variant.